Bump `Pkg` to latest tip of `Pkg.jl#master`

staticfloat opened the pull request:
Closes #38794

staticfloat committed:
Bump `Pkg` to latest tip of `Pkg.jl#master`

DilumAluthge commented:
Closes https://github.com/JuliaLang/julia/pull/38794?

fredrikekre commented:
Doctests in https://github.com/JuliaLang/Pkg.jl/blob/master/ext/LazilyInitializedFields/LazilyInitializedFields.jl fail. Should probably just make those `juila-repl`....

KristofferC commented:
Cherry pick https://github.com/JuliaLang/julia/pull/38794/commits/20c94f9d6e1143eb0fe0fd96df880c1be67a75c9.

staticfloat committed:
add a docmeta to Pkg

Forward-porting
https://github.com/JuliaLang/julia/commit/20c94f9d6e1143eb0fe0fd96df880c1be67a75c9
